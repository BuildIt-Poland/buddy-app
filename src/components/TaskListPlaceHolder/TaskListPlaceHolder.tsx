import React from 'react';
import makeStyles from '@material-ui/core/styles/makeStyles';
import Grey from '@material-ui/core/colors/grey';

const useStyles = makeStyles(theme => ({
  '@keyframes placeholder': {
    '0%': {
      backgroundPosition: '0% 50%',
    },
    '50%': { backgroundPosition: '100% 50%' },
    '100%': { backgroundPosition: '0% 50%' },
  },
  subHeader: {
    height: '4rem',
    width: '15rem',
    margin: theme.spacing(1, 0),
    background: `linear-gradient(89deg, ${theme.palette.background.paper}, ${Grey[300]})`,
    backgroundSize: '600% 600%',
    animation: '$placeholder 1s ease infinite',
  },
  listItem: {
    height: '4rem',
    margin: theme.spacing(1, 0),
    background: `linear-gradient(89deg, ${theme.palette.background.paper}, ${Grey[300]})`,
    backgroundSize: '600% 600%',
    animation: '$placeholder 1s ease infinite',
  },
}));

const PlaceHolderTaskList: React.FC = () => {
  const classes = useStyles();

  const PlaceHolderSubHeader = () => <div className={classes.subHeader} />;

  const PlaceHolderListItem = () => <div className={classes.listItem} />;

  return (
    <>
      <PlaceHolderSubHeader />
      <PlaceHolderListItem />
      <PlaceHolderListItem />
      <PlaceHolderListItem />
      <PlaceHolderSubHeader />
      <PlaceHolderListItem />
      <PlaceHolderListItem />
      <PlaceHolderListItem />
    </>
  );
};

export default PlaceHolderTaskList;
