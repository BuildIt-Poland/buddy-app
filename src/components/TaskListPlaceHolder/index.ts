import TaskListPlaceHolder from './TaskListPlaceHolder';

export default TaskListPlaceHolder;
