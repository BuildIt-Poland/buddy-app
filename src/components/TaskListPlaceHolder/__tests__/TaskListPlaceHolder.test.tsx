import React from 'react';
import { create } from 'react-test-renderer';

import TaskListPlaceHolder from '../TaskListPlaceHolder';

jest.mock('@material-ui/core/Paper', () => 'Paper');

describe('Component - TaskListPlaceHolder', () => {
  test('renders correctly', () => {
    const component = create(<TaskListPlaceHolder />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
