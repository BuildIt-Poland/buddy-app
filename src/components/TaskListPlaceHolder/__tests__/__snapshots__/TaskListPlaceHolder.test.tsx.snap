// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - TaskListPlaceHolder renders correctly 1`] = `
Array [
  <div
    className="makeStyles-subHeader-2"
  />,
  <div
    className="makeStyles-listItem-3"
  />,
  <div
    className="makeStyles-listItem-3"
  />,
  <div
    className="makeStyles-listItem-3"
  />,
  <div
    className="makeStyles-subHeader-2"
  />,
  <div
    className="makeStyles-listItem-3"
  />,
  <div
    className="makeStyles-listItem-3"
  />,
  <div
    className="makeStyles-listItem-3"
  />,
]
`;
